devicetypes/smartthings/zigbee-button.src/zigbee-button.groovy

```diff
@@ -14,181 +13,321 @@
+        }
+
+        valueTile("battery", "device.battery", decoration: "flat", inactiveLabel: false) {
+            state "battery", label:'${currentValue}% battery', unit:""
+        }
+
+        valueTile("temperature", "device.temperature", width: 2, height: 2) {
+            state("temperature", label:'${currentValue}°',
+                backgroundColors:[
+                    [value: 31, color: "#153591"],
+                    [value: 44, color: "#1e9cbb"],
+                    [value: 59, color: "#90d2a7"],
+                    [value: 74, color: "#44b621"],
+                    [value: 84, color: "#f1d801"],
+                    [value: 95, color: "#d04e00"],
+                    [value: 96, color: "#bc2323"]
+                ]
+            )
+        }
+
+        standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat") {
+            state "default", action:"refresh.refresh", icon:"st.secondary.refresh"
+        }
+        main (["button"])
+        details(["button", "battery", "temperature", "refresh"])
+    }
 }
 
 def parse(String description) {
-    def descMap = zigbee.parseDescriptionAsMap(description)
-    logIt descMap
-    state.lastCheckin = now()
-    logIt "lastCheckin = ${state.lastCheckin}"
-    handlePresenceEvent(true)
-    
-	def results = []
-    if (description?.startsWith('catchall:'))
-		results = parseCatchAllMessage(descMap)
-	else if (description?.startsWith('read attr -'))
-		results = parseReportAttributeMessage(descMap)
-    else logIt(descMap, "trace")
-        
-	return results;
-}
+    log.debug "description is $description"
+    def event = zigbee.getEvent(description)
+    if (event) {
+        sendEvent(event)
+    }
+    else {
+        if ((description?.startsWith("catchall:")) || (description?.startsWith("read attr -"))) {
+            def descMap = zigbee.parseDescriptionAsMap(description)
+            if (descMap.clusterInt == 0x0402 && descMap.attrInt == 0) {
+                event = parseTemperature(descMap.value)
+            }
+            else if (descMap.clusterInt == 0x0001 && descMap.attrInt == 0x0020) {
+                event = getBatteryResult(zigbee.convertHexToInt(descMap.value))
+            }
+            else if (descMap.clusterInt == 0x0006 || descMap.clusterInt == 0x0008) {
+                event = parseNonIasButtonMessage(descMap)
+            }
+        }
+        else if (description?.startsWith('temperature: ')) {
+            def value = zigbee.parseHATemperatureValue(description, "temperature: ", getTemperatureScale())
+            event = getTemperatureResult(value)
+        } else if (description?.startsWith('zone status')) {
+            event = parseIasButtonMessage(description)
+        }
 
-def updated() {
-	startTimer()
-    configure()
+        log.debug "Parse returned $event"
+        def result = event ? createEvent(event) : null
```

Comment:
I believe this should set result to an empty list instead of null to prevent an event from being created by the platform.